applications/StructuralMechanicsApplication/custom_elements/base_shell_element.cpp

```diff
@@ -0,0 +1,658 @@
+}
+
+void BaseShellElement::BaseFinalizeNonLinearIteration(ProcessInfo& rCurrentProcessInfo)
+{
+    const GeometryType& geom = this->GetGeometry();
+    const Matrix& shapeFunctionsValues = geom.ShapeFunctionsValues(GetIntegrationMethod());
+    for (SizeType i = 0; i < mSections.size(); ++i)
+        mSections[i]->FinalizeNonLinearIteration(GetProperties(), geom, row(shapeFunctionsValues, i), rCurrentProcessInfo);
+}
+
+void BaseShellElement::BaseInitializeSolutionStep(ProcessInfo& rCurrentProcessInfo)
+{
+	const PropertiesType& props = GetProperties();
+	const GeometryType & geom = GetGeometry();
+	const Matrix& shapeFunctionsValues = geom.ShapeFunctionsValues(GetIntegrationMethod());
+
+	for (SizeType i = 0; i < mSections.size(); ++i)
+		mSections[i]->InitializeSolutionStep(props, geom, row(shapeFunctionsValues, i), rCurrentProcessInfo);
+}
+
+void BaseShellElement::BaseFinalizeSolutionStep(ProcessInfo& rCurrentProcessInfo)
+{
+    const PropertiesType& props = GetProperties();
+    const GeometryType& geom = GetGeometry();
+    const Matrix& shapeFunctionsValues = geom.ShapeFunctionsValues(GetIntegrationMethod());
+
+    for (SizeType i = 0; i < mSections.size(); i++)
+        mSections[i]->FinalizeSolutionStep(props, geom, row(shapeFunctionsValues, i), rCurrentProcessInfo);
+}
+
+void BaseShellElement::CalculateLocalSystem(MatrixType& rLeftHandSideMatrix,
+	VectorType& rRightHandSideVector,
+	ProcessInfo& rCurrentProcessInfo)
+{
+    // Calculation flags
+    const bool calculate_stiffness_matrix_flag = true;
+    const bool calculate_residual_vector_flag = true;
+
+	CalculateAll(rLeftHandSideMatrix, rRightHandSideVector, rCurrentProcessInfo,
+                 calculate_stiffness_matrix_flag, calculate_residual_vector_flag);
+}
+
+void BaseShellElement::CalculateRightHandSide(VectorType& rRightHandSideVector,
+	ProcessInfo& rCurrentProcessInfo)
+{
+    // Calculation flags
+    const bool calculate_stiffness_matrix_flag = true; // TODO check is this can be false => see solids
+    const bool calculate_residual_vector_flag = true;
+
+	Matrix dummy;
+	CalculateAll(dummy, rRightHandSideVector, rCurrentProcessInfo,
+                 calculate_stiffness_matrix_flag, calculate_residual_vector_flag);
+}
+
+void BaseShellElement::CalculateMassMatrix(MatrixType& rMassMatrix, ProcessInfo& rCurrentProcessInfo)
+{
+    // TODO unify implementation and move it to BaseClass
+}
+
+void BaseShellElement::CalculateDampingMatrix(
+    MatrixType& rDampingMatrix,
+    ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_TRY;
+
+    const SizeType num_dofs = GetNumberOfDofs();
+
+    if ( rDampingMatrix.size1() != num_dofs )
+        rDampingMatrix.resize( num_dofs, num_dofs, false );
+
+    noalias( rDampingMatrix ) = ZeroMatrix( num_dofs, num_dofs );
+
+    // 1.-Calculate StiffnessMatrix:
+
+    MatrixType StiffnessMatrix  = Matrix();
```

Comment:
you don't need to change, but note that

    MatrixType StiffnessMatrix;

would have been equivalent